multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/DetectDeployedMtaStep.java

```diff
@@ -51,7 +52,11 @@ private DeployedMta detectDeployedMta(String mtaId, String mtaNamespace, CloudCo
         Optional<DeployedMta> optionalDeployedMta = deployedMtaDetector.detectDeployedMtaByNameAndNamespace(mtaId, mtaNamespace, client);
 
         if (optionalDeployedMta.isEmpty()) {
-            getStepLogger().info(Messages.NO_DEPLOYED_MTA_DETECTED);
+            if (StringUtils.isNotEmpty(mtaNamespace)) {
+                getStepLogger().info(MessageFormat.format(Messages.NO_DEPLOYED_MTA_DETECTED_WITH_NAMESPACE, mtaId, mtaNamespace));
+            } else {
+                getStepLogger().info(MessageFormat.format(Messages.NO_DEPLOYED_MTA_DETECTED, mtaId));
+            }
```

Comment:
Extract in new method and remove else statement(just exit from method if statement is true)